tycho-packaging-plugin/src/main/java/org/eclipse/tycho/packaging/AbstractTychoPackagingMojo.java

```diff
@@ -172,6 +173,10 @@ protected MavenProject updatePom(String finalName) throws IOException {
 		for (Dependency dep : list) {
 			Dependency copy = dep.clone();
 			copy.setSystemPath(null);
+			if (dep.getGroupId().startsWith("p2.eclipse.") && Artifact.SCOPE_SYSTEM.equals(dep.getScope())) {
+				copy.setOptional(true);
+				copy.setScope(Artifact.SCOPE_PROVIDED);
+			}
```

Comment:
Isn't it/shouldn't it be part of a separate commit?
Couldn't it be then that instead of a repository id, we could then assume that everything that's not "provided" comes either from reactor or repository?

Reply:
> Isn't it/shouldn't it be part of a separate commit?
Maybe yes, I just want to make the big picture work here :-)

> Couldn't it be then that instead of a repository id, we could then assume that everything that's not "provided" comes either from reactor or repository?

I hope we can over the time make more and more items with the real scope, and maybe provided would become a valid scope then also even though we currently only use compile+test+system, so my goal is to be as specific as possible here.